Automattic/wp-calypso
# Downtime Montoring SMS add phone number list to modal

Related to # 1204774821045518-as-1204784983353063/f

## Proposed Changes

* This PR adds a new component to the SMS section in the settings modal
* The component displays all of the SMS numbers and names, similar to the email field
* Right now we're using a hardcoded array of sms number objects pending the API implementation
* I've also added the context menu for additional numbers, but none of the items are functional
* It seems to me that the CSS can be unified between the email and SMS cards, but I didn't do that now in case there are differences between the two I'm not yet aware of
* I left the default contact functionality that's found in email in place, but I don't know if there will be an actual default SMS contact, so that bit can be removed if necessary

![Screenshot 2023-06-08 at 10 27 15 AM](https://github.com/Automattic/wp-calypso/assets/12895386/0d3f8b98-1765-434b-b2ca-607da92861cd)

![Screenshot 2023-06-08 at 10 27 40 AM](https://github.com/Automattic/wp-calypso/assets/12895386/7d2e5e58-394a-47ad-942c-9d24ca6c5263)


## Testing Instructions

* checkout this PR locally and fire up yarn start-jetpack-cloud
* Make sure you have an agency partner account and head over to jetpack.cloud.localhost:3000 
* Click the notification settings icon next to one of your sites
* If it's not toggled on, toggle on SMS notification
* Observe there is now two dummy contacts in the list
* Click the elipse to the right of the 2nd contact to see the context menu
* Enable email to verify that it still correctly shows email contacts for the site

## Pre-merge Checklist
